### PR TITLE
Support for card rarities in Set Draft mode + many bug fixes (involving set/deck editing, flavor text, etc)

### LINKED ISSUE #1532: Set Draft: support card rarities in sets

### LINKED ISSUE #1656: Set being edited doesn't actually get loaded when refreshing a /set/[...] page

### LINKED ISSUE #1657: Deck name doesn't load when refreshing a /deck/[...] edit page

### LINKED ISSUE #1658: Flavor text on cards doesn't get saved correctly

### LINKED ISSUE #1659: Flavor text on cards doesn't display in CardGrid view

### LINKED ISSUE #1660: Flavor text on cards doesn't load when editing a card with flavor text in the Workshop

### LINKED ISSUE #1661: Rendered cards overflow slightly (in the stats area) when viewed at scale<1 (i.e. in set summary view)

### LINKED ISSUE #1662: Make flavor text on cards easier to read (toggle rather than tooltip)

### LINKED ISSUE #1663: Aborting a set draft by refreshing a page doesn't abort it for the other player

### LINKED ISSUE #1664: Aborted/forfeited games don't log a message in the game log about the abort/forfeit

### LINKED ISSUE #1665: "Preview" in the Workshop text blends into buttons above it at smaller screen widths (<1250px)

### LINKED ISSUE #1667: Increase minimum set size from 15 to 20

### PR DESCRIPTION
**Pull request by AlexNisnevich**
Rarities for cards in sets (close #1532):
* Sets can optionally define rarities (common, uncommon, rare) for cards in that set. Either all cards or no cards in a given set can have rarities.
* Rarities affect only the Set Draft format – they determine how often cards come up (on avg: 3 rare cards, 7 uncommon cards, 20 common cards per drafted deck of 30)
* Cards display their rarities only when viewing/editing a set with rarities or drafting/playing in the Set Draft format
* Corresponding documentation and validation (sets with card rarities enabled must have ≥4 cards of each rarity)

Bug fixes unrelated to card rarities:
* fix #1656
* fix #1657
* fix #1658
* fix #1659
* fix #1660
* fix #1661
* fix #1662
* fix #1663
* fix #1664
* fix #1665
* fix #1667

### CODE REVIEW

File: src/common/components/card/RaritySymbol.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+export default function RaritySymbol({ rarity, scale, isEditing }: RaritySymbolProps): JSX.Element {
+  return (
+    <span style={{
+      display: 'inline-block',
+      userSelect: 'none',
+      fontSize: ({ rare: 17, uncommon: 27, common: 26 })[rarity] * (scale || 1),
+      marginTop: ({ rare: 5, uncommon: -3, common: 0 })[rarity] * (scale || 1),
+      marginLeft: ({ rare: -1, uncommon: -2, common: 0 })[rarity] * (scale || 1),
+      color: ({ rare: '#D4AF37', uncommon: '#aaa9ad', common: undefined })[rarity]
```
Comment (jacobnisnevich) [inline src/common/components/card/RaritySymbol.tsx:21]: Why are some hexcodes all caps and other all lower?

File: src/common/components/cards/ActiveDeck.tsx [hunk 2]
```diff
@@ -72,12 +77,27 @@ export default class ActiveDeck extends React.Component<ActiveDeckProps, ActiveD
   get hasRightCardCount(): boolean {
     const { cards, isASet } = this.props;
     if (isASet) {
-      return cards.length >= 15;
+      return cards.length >= 20;
```
Comment (jacobnisnevich) [inline src/common/components/cards/ActiveDeck.tsx:80]: Maybe pull these out as constants to more easily configure?

File: src/common/components/cards/ActiveDeckCard.tsx [hunk 3]
```diff
@@ -54,40 +58,64 @@ export default class ActiveDeckCard extends React.Component<ActiveDeckCardProps>
...
     }
   };
 
   public render(): JSX.Element {
-    const { card, showCount } = this.props;
+    const { card, showCount, rarity } = this.props;
     return (
-      <div>
-        <span style={this.styles.xButton} onClick={this.handleRemoveCard}>
+      <div style={{ display: 'flex', justifyContent: 'spaceBetween' }}>
```
Comment (jacobnisnevich) [inline src/common/components/cards/ActiveDeckCard.tsx:77]: Pull this out into `styles`?

File: src/common/util/sets.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        ? 4
+        : 3;
+
+  // minimum: 15 - 5   - 3   =  7 groups (= 14 cards = 46 2/3%)
+  // average: 15 - 3.5 - 1.5 = 10 groups (= 20 cards = 66 2/3%)
+  // maximum: 15 - 3   - 1   = 11 groups (= 22 cards = 73 1/3%)
+  const numCommonGroups = totalNumGroups - numUncommonGroups - numRareGroups;
+
+  return shuffle([
+    ...new Array(numRareGroups).fill('rare'),
```
Comment (jacobnisnevich) [inline src/common/util/sets.ts:89]: Nit: constants for `rare`, `uncommon`, `common`

Comment (jacobnisnevich) [reply src/common/util/sets.ts:89]: Better yet a const enum

**Review by jacobnisnevich — APPROVED** [review]

File: src/common/util/sets.ts (hunk 4 above)
Comment (AlexNisnevich) [reply src/common/util/sets.ts:89]: eh, the typechecker already kind of makes sure things are fine here

**Pull request merged**